src/prepare.js

```diff
@@ -179,6 +186,113 @@ function getConfigOverride(confStr) {
 	return config;
 }
 
+/**
+ * @param {ReturnType<getConfigOverride>} conf
+ * @param {ReturnType<typeof getBasicBuildOptions>["toolchain"]} toolchain
+ */
+async function extendConfigForW3CDeploy(conf, toolchain) {
+	/** Get present date in YYYY-MM-DD format */
+	const getShortIsoDate = () => new Date().toISOString().slice(0, 10);
+
+	let publishDate = getShortIsoDate();
+	if (toolchain === "respec") {
+		conf.publishDate = publishDate = conf.publishDate || publishDate;
+	} else if (toolchain === "bikeshed") {
+		conf.date = publishDate = conf.date || publishDate;
+	}
+
+	const shortName = conf.shortName || conf.shortname;
+	if (shortName) {
```

Comment:
In progress (next PR): parsing `shortName` from `SOURCE` if user has not provided any in `W3C_BUILD_OVERRIDE`.

Reply:
@dontcallmedom Is `w3c.json` a reliable source to get `shortName`, or do I need to parse it from input file?

Reply:
no, w3c.json is not a reliable source to get the shortname; that said, contrarily to status and previous versions, the shortname changes very rarely so to me, it would be more than OK to require it in the action config if parsing it is too brittle or cumbersome.

Reply:
Parsing respecConfig is straight-forward. I can open source with puppeteer and read `window.respecConfig.shortName`.
With Bikeshed, I need to process the source file manually, which includes parsing `<pre class='metadata'>` using regex followed by parsing key-value pairs, which follow same format as W3C_BUILD_OVERRIDE.